pkg/subvolume/subvolume.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subvolume
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	exec "github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type fsStruct struct {
+	Name string
+	data []string
+}
+
+type subVolume struct {
+	fs     string
+	svg    string
+	subvol string
+}
+
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace, flag string) []subVolume {
+
+	pvList, err := clientsets.Kube.CoreV1().PersistentVolumes().List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching PVs: %v\n", err))
+		os.Exit(1)
+	}
+	var subvolumeNames []string
+	for _, pv := range pvList.Items {
+		if pv.Spec.CSI.VolumeAttributes["subvolumeName"] != "" {
+			subvolumeNames = append(subvolumeNames, pv.Spec.CSI.VolumeAttributes["subvolumeName"])
+		}
+	}
+
+	sv := getCephFSSubvolumes(ctx, clientsets, clusterNamespace, operatorNamespace, flag)
```

Comment:
Let's pass the subvolumeNames map directly into this helper. Then as each stale or non-stale subvolume is found, it can be immediately printed. Then there is no need to wait until the end of the processing. Otherwise, if nothing is printed until the end, they will think it hung, especially if there are many volumes.
```suggestion
	sv := getCephFSSubvolumes(ctx, clientsets, clusterNamespace, operatorNamespace, subvolumeNames, flag)
```